### PR TITLE
Fix deferred course start date on offer deferred

### PR DESCRIPTION
**Pull request by CatalinVoineag**
## Context

Previously, it didn't show the actual date of the course start when it's deferred.

It was showing the date of the current cycle, not the next. So 2024 not 2025 for example

## Changes proposed in this pull request

Added deferred_start_date

## Guidance to review

Is the logic of the method correct?
Go on the review app and sign in as a candidate with a deferred offer and check the start date of your course is correct, in the next cycle, should be 2025 for a deferred offer in 2024 cycle 

https://github.com/user-attachments/assets/6358bd57-0458-40da-8e4f-0617fda191f3

## Things to check

- [ ] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [ ] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] If this code adds a column to the DB, decide whether it needs to be in analytics yml file or analytics blocklist
- [ ] API release notes have been updated if necessary
- [ ] If it adds a significant user-facing change, is it documented in the [CHANGELOG](CHANGELOG.md)?
- [ ] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)
- [ ] Inform data insights team due to database changes
- [x] Make sure all information from the Trello card is in here
- [x] Rebased main
- [x] Cleaned commit history
- [x] Tested by running locally
- [x] Add PR link to Trello card

### CODE REVIEW

**Comment by CatalinVoineag** [comment]
[Incorrect course start date for deferred offer](https://trello.com/c/osHvS3wj/2373-incorrect-course-start-date-for-deferred-offer)

**Review by inulty-dfe — CHANGES_REQUESTED** [review]
If the application has been deferred, there exists a course in the next cycle with a start date, right? Would it be better to get the start date for that course?

```
capplication_choice.current_course_option.course.start_date
```

**Review by elceebee — COMMENTED** [review]
We render this same information in the [application_status_tag_component](https://github.com/DFE-Digital/apply-for-teacher-training/blob/9f8a2a395bc477ff8c004735ec069f24d92c8686/app/components/candidate_interface/application_status_tag_component.html.erb).

It is correctly rendered there. 

I'm happy with the method you've written, or Iain's suggestion (I don't feel strongly about it). But it would be good to use the same method in both places.

File: app/components/candidate_interface/application_status_tag_component.html.erb [hunk 1]
```diff
@@ -40,7 +40,7 @@
     </p>
   <% elsif  @application_choice.offer_deferred? %>
     <p class="govuk-body govuk-!-margin-top-2">
-      Your training will now start in <%= (@application_choice.current_course_option.course.start_date + 1.year).to_fs(:month_and_year) %>.
+      Your training will now start in <%= @application_choice.current_course.deferred_start_date.to_fs(:month_and_year) %>.
```
Comment (CatalinVoineag) [inline app/components/candidate_interface/application_status_tag_component.html.erb:43]: current_course_option.course and current_course are the same thing I think https://github.com/DFE-Digital/apply-for-teacher-training/blob/main/app/models/application_choice.rb#L20

Comment (inulty-dfe) [reply app/components/candidate_interface/application_status_tag_component.html.erb:43]: Ah, I'm thinking of confirmed deferrals. This is good.

Comment (CatalinVoineag) [reply app/components/candidate_interface/application_status_tag_component.html.erb:43]: When you confirm a deferral the application choice status changes to recruited so this code won't be called then

**Review by inulty-dfe — APPROVED** [review]
:rocket:

**Pull request merged**